### ISSUE TITLE
Cinnamon 5.4 - keyboard doesn't work after login

### ISSUE THREAD

**Issue by Rizado** [linuxmint/cinnamon#10831]
```
 * Cinnamon version 5.4.0
 * Distribution - Mint 21
 * Intel UHD Graphics 630 (internal graphic core of CPU - i5-10500)
 * 64 bit
 * Attach ~/.xsession-errors, or /var/log/syslog
 ```
dbus-update-activation-environment: setting DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
dbus-update-activation-environment: setting DISPLAY=:0
dbus-update-activation-environment: setting XAUTHORITY=/home/rizado/.Xauthority
/etc/X11/Xsession.d/30x11-common_xresources: строка 16: has_option: команда не найдена
/etc/X11/Xsession.d/75dbus_dbus-launch: строка 9: has_option: команда не найдена
dbus-update-activation-environment: setting GTK_MODULES=gail:atk-bridge
dbus-update-activation-environment: setting QT_ACCESSIBILITY=1
/etc/X11/Xsession.d/90x11-common_ssh-agent: строка 9: has_option: команда не найдена
dbus-update-activation-environment: setting SHELL=/bin/bash
dbus-update-activation-environment: setting QT_ACCESSIBILITY=1
dbus-update-activation-environment: setting XDG_CONFIG_DIRS=/etc/xdg/xdg-cinnamon:/etc/xdg
dbus-update-activation-environment: setting XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
dbus-update-activation-environment: setting LANGUAGE=ru_RU
dbus-update-activation-environment: setting LC_ADDRESS=ru_RU.UTF-8
dbus-update-activation-environment: setting LC_NAME=ru_RU.UTF-8
dbus-update-activation-environment: setting DESKTOP_SESSION=cinnamon
dbus-update-activation-environment: setting LC_MONETARY=ru_RU.UTF-8
dbus-update-activation-environment: setting GTK_MODULES=gail:atk-bridge
dbus-update-activation-environment: setting PWD=/home/rizado
dbus-update-activation-environment: setting XDG_SESSION_DESKTOP=cinnamon
dbus-update-activation-environment: setting LOGNAME=rizado
dbus-update-activation-environment: setting XDG_SESSION_TYPE=x11
dbus-update-activation-environment: setting GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1
dbus-update-activation-environment: setting XAUTHORITY=/home/rizado/.Xauthority
dbus-update-activation-environment: setting XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/rizado
dbus-update-activation-environment: setting GDM_LANG=ru
dbus-update-activation-environment: setting HOME=/home/rizado
dbus-update-activation-environment: setting IM_CONFIG_PHASE=1
dbus-update-activation-environment: setting LANG=ru_RU.UTF-8
dbus-update-activation-environment: setting LC_PAPER=ru_RU.UTF-8
dbus-update-activation-environment: setting XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
dbus-update-activation-environment: setting XDG_SESSION_CLASS=user
dbus-update-activation-environment: setting LC_IDENTIFICATION=ru_RU.UTF-8
dbus-update-activation-environment: setting USER=rizado
dbus-update-activation-environment: setting DISPLAY=:0
dbus-update-activation-environment: setting SHLVL=1
dbus-update-activation-environment: setting LC_TELEPHONE=ru_RU.UTF-8
dbus-update-activation-environment: setting LC_MEASUREMENT=ru_RU.UTF-8
dbus-update-activation-environment: setting PAPERSIZE=a4
dbus-update-activation-environment: setting XDG_RUNTIME_DIR=/run/user/1000
dbus-update-activation-environment: setting LC_TIME=ru_RU.UTF-8
dbus-update-activation-environment: setting GTK3_MODULES=xapp-gtk3-module
dbus-update-activation-environment: setting XDG_DATA_DIRS=/usr/share/cinnamon:/usr/share/gnome:/home/rizado/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share
dbus-update-activation-environment: setting PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
dbus-update-activation-environment: setting GDMSESSION=cinnamon
dbus-update-activation-environment: setting DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
dbus-update-activation-environment: setting LC_NUMERIC=ru_RU.UTF-8
dbus-update-activation-environment: setting _=/usr/bin/dbus-update-activation-environment
[cinnamon-settings-daemon-smartcard] Failed to start: не удалось найти подходящий драйвер смарт-карт
[cinnamon-settings-daemon-smartcard] Failed to start: не удалось найти подходящий драйвер смарт-карт
mutter-Message: 21:40:55.583: Enabling experimental feature 'x11-randr-fractional-scaling'

(cinnamon:1831): mutter-WARNING **: 21:40:55.709: Failed to migrate old monitors config file: Ошибка в строке 14 на символе 27: Expected a number, got 29,969000
Gjs-Message: 21:40:55.766: Profiler is disabled. Not setting up signals.
Gjs-Message: 21:40:55.893: JS LOG: About to start Cinnamon
Gjs-Message: 21:40:55.984: JS LOG: [LookingGlass/info] Cinnamon.AppSystem.get_default() started in 90 ms
Gjs-Message: 21:40:55.987: JS LOG: [LookingGlass/info] loading user theme: /home/rizado/.themes/Mint-Y-Dark-Perso/cinnamon/cinnamon.css
Gjs-Message: 21:40:55.997: JS LOG: [LookingGlass/info] added icon directory: /home/rizado/.themes/Mint-Y-Dark-Perso/cinnamon
Gjs-Message: 21:40:56.093: JS LOG: [LookingGlass/info] loaded at Sun Jun 12 2022 21:40:56 GMT+0300 (Москва, стандартное время)
Gjs-Message: 21:40:56.093: JS LOG: Cinnamon started at Sun Jun 12 2022 21:40:56 GMT+0300 (Москва, стандартное время)
Gjs-Message: 21:40:56.093: JS LOG: GPU offload supported: false
Gjs-Message: 21:40:56.100: JS LOG: [LookingGlass/info] DeskletManager started in 1 ms
Gjs-Message: 21:40:56.100: JS LOG: [LookingGlass/info] SearchProviderManager started in 1 ms
Gjs-Message: 21:40:56.190: JS LOG: [LookingGlass/info] Loaded applet removable-drives@cinnamon.org in 38 ms
Gjs-Message: 21:40:56.193: JS LOG: [LookingGlass/info] Role locked: tray
Gjs-Message: 21:40:56.197: JS LOG: [LookingGlass/info] Loaded applet systray@cinnamon.org in 7 ms
Gjs-Message: 21:40:56.252: JS LOG: [LookingGlass/info] [weather@mockturtl#13]: Internet access now available, resuming operations.
Gjs-Message: 21:40:56.269: JS LOG: [LookingGlass/info] Loaded applet weather@mockturtl in 72 ms
Gjs-Message: 21:40:56.276: JS LOG: [LookingGlass/info] Loaded applet show-desktop@cinnamon.org in 7 ms

** (nemo-desktop:1899): WARNING **: 21:40:56.286: Current gtk theme is not known to have nemo support (Adwaita) - checking...
Gjs-Message: 21:40:56.330: JS LOG: [LookingGlass/info] Loaded applet printers@cinnamon.org in 54 ms

** (nemo-desktop:1899): WARNING **: 21:40:56.333: The theme appears to have no nemo support.  Adding some...
Gjs-Message: 21:40:56.337: JS LOG: [LookingGlass/info] Role locked: notifications
Gjs-Message: 21:40:56.348: JS LOG: [LookingGlass/info] Loaded applet notifications@cinnamon.org in 18 ms
Gjs-Message: 21:40:56.379: JS LOG: [LookingGlass/info] Loaded applet calendar@cinnamon.org in 31 ms
Gjs-Message: 21:40:56.385: JS LOG: [LookingGlass/info] Loaded applet workspace-switcher@cinnamon.org in 6 ms
Gjs-Message: 21:40:56.402: JS LOG: [LookingGlass/info] Loaded applet power@cinnamon.org in 17 ms
Gjs-Message: 21:40:56.403: JS LOG: [LookingGlass/info] Loaded extension DesktopCube@yare in 1 ms
Gjs-Message: 21:40:56.404: JS LOG: [LookingGlass/info] ExtensionSystem started in 304 ms
cinnamon-session[1457]: WARNING: t+1,17210s: Detected that screensaver has appeared on the bus
Gjs-Message: 21:40:56.420: JS LOG: [LookingGlass/info] Role locked: panellauncher
Gjs-Message: 21:40:56.423: JS LOG: [LookingGlass/info] Loaded applet grouped-window-list@cinnamon.org in 20 ms
Gjs-Message: 21:40:56.423: JS LOG: calendar@cinnamon.org: No calendar event support (needs evolution-data-server)

** (csd-power:1710): CRITICAL **: 21:40:56.430: abs_to_percentage: assertion 'max > min' failed
Gjs-Message: 21:40:56.446: JS LOG: [LookingGlass/info] Loaded applet sound@cinnamon.org in 23 ms
Gjs-Message: 21:40:56.463: JS LOG: [LookingGlass/info] Loaded applet network@cinnamon.org in 17 ms
** Message: 21:40:56.604: nemo-desktop: session is cinnamon, establishing proxy
Gjs-Message: 21:40:56.679: JS LOG: [LookingGlass/info] Loaded applet CinnVIIStarkMenu@NikoKrause in 216 ms
Gjs-Message: 21:40:56.679: JS LOG: [LookingGlass/info] AppletManager started in 586 ms
Gjs-Message: 21:40:56.681: JS LOG: [LookingGlass/info] Cinnamon took 788 ms to start
Gjs-Message: 21:40:56.688: JS LOG: Unknown network device type, is 14
Gjs-Message: 21:40:56.702: JS LOG: [LookingGlass/info] Hiding systray: network
Gjs-Message: 21:40:56.703: JS LOG: [LookingGlass/info] Adding systray: gxkb (16x16px)

(cinnamon:1831): St-WARNING **: 21:40:56.724: Ignoring length property that isn't a number at line 953, col 11
Gjs-Message: 21:40:58.619: JS LOG: [LookingGlass/info] [weather@mockturtl#13]: Weather Information refreshed
/usr/lib/linuxmint/mintUpdate/mintUpdate.py:1285: DeprecationWarning: Gdk.threads_init is deprecated
  Gdk.threads_init()
/usr/lib/linuxmint/mintUpdate/mintUpdate.py:1433: DeprecationWarning: Gtk.ImageMenuItem.set_image is deprecated
  menuItem3.set_image(image)
/usr/lib/linuxmint/mintUpdate/mintUpdate.py:733: DeprecationWarning: Gdk.threads_enter is deprecated
  Gdk.threads_enter()
/usr/lib/linuxmint/mintUpdate/mintUpdate.py:740: DeprecationWarning: Gdk.threads_leave is deprecated
  Gdk.threads_leave()
Gjs-Message: 21:41:16.333: JS LOG: [LookingGlass/info] Adding systray: mintupdate.py (1x1px)

(xapp-sn-watcher:1884): GLib-GObject-CRITICAL **: 21:41:26.643: g_object_set: assertion 'G_IS_OBJECT (object)' failed

(xapp-sn-watcher:1884): GLib-GIO-CRITICAL **: 21:41:26.643: g_dbus_interface_skeleton_flush: assertion 'G_IS_DBUS_INTERFACE_SKELETON (interface_)' failed

(xapp-sn-watcher:1884): GLib-GObject-CRITICAL **: 21:41:26.643: g_object_set: assertion 'G_IS_OBJECT (object)' failed

(xapp-sn-watcher:1884): GLib-GIO-CRITICAL **: 21:41:26.643: g_dbus_interface_skeleton_flush: assertion 'G_IS_DBUS_INTERFACE_SKELETON (interface_)' failed

(xapp-sn-watcher:1884): GLib-GObject-WARNING **: 21:41:26.643: invalid (NULL) pointer instance

(xapp-sn-watcher:1884): GLib-GObject-CRITICAL **: 21:41:26.643: g_signal_emit_by_name: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
cinnamon-session[1457]: WARNING: t+31,17561s: Detected that screensaver has left the bus

(cinnamon:1831): St-WARNING **: 21:41:26.932: Ignoring length property that isn't a number at line 953, col 11

(cinnamon:1831): St-WARNING **: 21:41:26.946: Ignoring length property that isn't a number at line 953, col 11

(cinnamon:1831): Gjs-WARNING **: 21:52:05.572: Some code tried to set a deprecated GObject property.
0 _adaptClones() ["/usr/share/cinnamon/js/ui/appSwitcher/timelineSwitcher.js":27:12]
1 _createList() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher3D.js":218:13]
2 _show() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher3D.js":67:13]
3 _show() ["self-hosted":850:0]

(cinnamon:1831): Gjs-WARNING **: 21:52:05.589: Some code tried to set a deprecated GObject property.
0 _hidePreviews() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher3D.js":112:40]
1 _hide() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher3D.js":131:17]
2 destroy() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher.js":439:17]
3 _activateSelected() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher.js":421:17]
4 _keyReleaseEvent() ["/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher.js":328:17]
5 _keyReleaseEvent() ["self-hosted":850:0]

**Issue**

After login keyboard doesn't work till I click with mouse. After first click keyboard works fine. Tested at 2 PC.

**Steps to reproduce**

Install all updates, reboot and login into system. Try to press Alt+F2, Win, etc...

**Expected behaviour**

Keyboard has to work immediately after login.

**Other information**

**Configuration info, if applicable**
If this bug report is related to an Applet, Desklet or Extension, please paste (or use a pastebin service) the offending extension's settings, if it has any.  You can obtain this by opening its configuration, clicking the 'hamburger' button in the upper-right corner of the window, and selecting "Export to a file". **Please be sure to review the contents and remove any personal data it may contain.**

```

Please paste here

```

**Comment by Rizado** [comment]
I installed new instance of Mint 20.3 at my PC, replaced repositories in /etc/apt/sources.list.d/... (focal -> jammy, una -> vanessa) and executed **sudo apt update && sudo apt dist-upgrade**. Problem still exists. After that I installed MATE. When I login into MATE session, keyboard works immediately fine. But in Cinnamon session keyboard works only after mouse click.

**Comment by Rizado** [comment]
@vladimir-didenko , please try to run Terminal using Ctrl+Alt+T (or your shortcut if you use different). Any keys and shortcuts don't work at my system.

**Comment by mtwebster** [comment]
I'm pretty sure this is fixed in git master:
https://github.com/linuxmint/muffin/commit/548ab2e864b703a63a3ef8a9429e5554b0a9399c

**Comment by Rizado** [comment]
@mtwebster , I'll reply you when this update will be in Linux Mint repository. And this fixes problem with Win (Meta) key only or with any hotkey? Ctrl+Alt+T (Terminal) doesn't work too.

**Comment by Rizado** [comment]
This problem is fixed now, in Cinnamon 5.4.1. Thanks!
I close the issue.
